File: sample-data-repository/src/test/java/nl/haploid/octowight/sample/TestData.java
package nl.haploid.octowight.sample;

import nl.haploid.octowight.registry.data.ResourceRoot;
import nl.haploid.octowight.sample.repository.PersonDmo;
import nl.haploid.octowight.sample.repository.RoleDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static String name() {
		return nextString();
	}

	public static String nextString() {
		return UUID.randomUUID().toString();
	}

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static PersonDmo personDmo() {
		final PersonDmo dmo = new PersonDmo();
		dmo.setId(nextLong());
		dmo.setName(name());
		return dmo;
	}

	public static RoleDmo roleDmo(final PersonDmo personDmo, final String type) {
		final RoleDmo dmo = new RoleDmo();
		dmo.setId(nextLong());
		dmo.setPerson(personDmo);
		dmo.setType(type);
		return dmo;
	}

	public static ResourceRoot resourceRoot() {
		final ResourceRoot resourceRoot = new ResourceRoot();
		resourceRoot.setAtomId(nextLong());
		resourceRoot.setAtomOrigin(nextString());
		resourceRoot.setAtomType(nextString());
		resourceRoot.setResourceId(nextLong());
		resourceRoot.setResourceType(nextString());
		resourceRoot.setVersion(nextLong());
		return resourceRoot;
	}
}

File: sample-data-repository/src/test/scala/nl/haploid/octowight/sample/TestData.scala
package nl.haploid.octowight.sample

import java.util.{Random, UUID}

import nl.haploid.octowight.registry.data.{Atom, ResourceRoot}
import nl.haploid.octowight.sample.repository.{PersonDmo, RoleDmo}

object TestData {

  def name = nextString

  def nextString = UUID.randomUUID.toString

  def nextLong: Long = new Random().nextLong

  def personDmo = {
    val dmo = new PersonDmo
    dmo.setId(nextLong)
    dmo.setName(name)
    dmo
  }

  def roleDmo(personDmo: PersonDmo, name: String) = {
    val dmo = new RoleDmo
    dmo.setId(nextLong)
    dmo.setPerson(personDmo)
    dmo.setName(name)
    dmo
  }

  def resourceRoot = {
    val resourceRoot = new ResourceRoot
    resourceRoot.setRoot(new Atom(nextLong, nextString, nextString))
    resourceRoot.setResourceId(nextLong)
    resourceRoot.setResourceType(nextString)
    resourceRoot.setVersion(nextLong)
    resourceRoot
  }
}
